forc-plugins/forc-client/src/op/call.rs

```diff
@@ -0,0 +1,2363 @@
+                ));
+                let tx = call
+                    .with_external_contracts(external_contracts)
+                    .transaction_builder(tx_policies, variable_output_policy, &wallet)
+                    .await
+                    .expect("Failed to build transaction")
+                    .with_build_strategy(ScriptBuildStrategy::StateReadOnly)
+                    .build(provider)
+                    .await?;
+                let tx_hash = tx.id(provider.chain_id());
+                let gas_price = gas.map(|g| g.price).unwrap_or(Some(0));
+                let tx_status = provider
+                    .dry_run_opt(tx, false, gas_price)
+                    .await
+                    .expect("Failed to simulate transaction");
+                (tx_status, tx_hash)
+            }
+            cmd::call::ExecutionMode::Live => {
+                forc_tracing::println_action_green(
+                    "Sending transaction with wallet",
+                    &format!("0x{}", wallet.address().hash()),
+                );
+                let tx = call
+                    .with_external_contracts(external_contracts)
+                    .build_tx(tx_policies, variable_output_policy, &wallet)
+                    .await
+                    .expect("Failed to build transaction");
+                let tx_hash = tx.id(provider.chain_id());
+                let tx_status = provider
+                    .send_transaction_and_await_commit(tx)
+                    .await
+                    .expect("Failed to send transaction");
+                (tx_status, tx_hash)
+            }
+        };
+
+        let receipts = tx_status
+            .take_receipts_checked(Some(&log_decoder))
+            .expect("Failed to take receipts");
+
+        let data = ReceiptParser::new(&receipts, DecoderConfig::default())
+            .extract_contract_call_data(contract_id)
+            .expect("Failed to extract contract call data");
+        let token = ABIDecoder::default()
+            .decode(&output_param, &data)
+            .expect("Failed to decode output");
+
+        let result = token_to_string(&token).expect("Failed to convert token to string");
+
+        forc_tracing::println_action_green("receipts:", &format!("{:#?}", receipts));
+        forc_tracing::println_action_green("tx hash:", &tx_hash.to_string());
+        forc_tracing::println_action_green("result:", &result);
+
+        // display transaction url if live mode
+        if cmd::call::ExecutionMode::Live == mode {
+            if let Some(explorer_url) = get_explorer_url(&node) {
+                forc_tracing::println_action_green(
+                    "\nView transaction:",
+                    &format!("{}/tx/0x{}", explorer_url, tx_hash),
+                );
+            }
+        }
+
+        return Ok(result);
+    }
+
+    let cmd::call::FuncType::Signature(function_signature) = function else {
+        bail!("Function must be a signature if no ABI is provided");
```

Comment:
nit: It would be good to give an example here. 
```suggestion
        bail!("Function must be a signature if no ABI is provided, e.g. "transfer(address,u64)"
```

I wasn't able to get it to work without providing the ABI. These commands I tried all gave me panics:
- `forc call --testnet 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter(1)"`
- `forc call --testnet 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter(1, u64)"`
- `forc call --testnet 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter(value,u64) 1"`
- `forc call --testnet 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter(value,u64)" 1`
- `forc call --testnet 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter(value:u64)" 1`

This worked however: `forc call --testnet --abi examples/counter/out/debug/counter-abi.json 0xd8b98e8ceebf08e2fbc4bf0599e8eae239608636b9aab76b5348c2174d197848 "initialize_counter" 1 `

Reply:
Got it, we should just make sure the CLI isn't panicking in any scenario